nixos/modules/services/monitoring/cockpit.nix

```diff
@@ -0,0 +1,225 @@
+        SocketMode="0600";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-https@.service
+    systemd.services."cockpit-wsinstance-https@" = {
+      description = "Cockpit Web Service https instance %I";
+      bindsTo = [ "cockpit.service"];
+      path = [ cfg.package ];
+      documentation = [ "man:cockpit-ws(8)" ];
+      serviceConfig = {
+        Slice = "system-cockpithttps.slice";
+        ExecStart = "${cfg.package}/libexec/cockpit-ws --for-tls-proxy --port=0";
+        User = "root";
+        Group = "";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-http.socket
+    systemd.sockets.cockpit-wsinstance-http = {
+      unitConfig = {
+        Description="Socket for Cockpit Web Service http instance";
+        BindsTo="cockpit.service";
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/http.sock";
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-https-factory.socket
+    systemd.sockets.cockpit-wsinstance-https-factory = {
+      unitConfig = {
+        Description="Socket for Cockpit Web Service https instance factory";
+        BindsTo="cockpit.service";
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/https-factory.sock";
+        Accept=true;
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-https-factory@.service
+    systemd.services."cockpit-wsinstance-https-factory@" = {
+      description = "Cockpit Web Service https instance factory";
+      documentation = [ "man:cockpit-ws(8)" ];
+      path = [ cfg.package ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/libexec/cockpit-wsinstance-factory";
+        User = "root";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-http.service
+    systemd.services."cockpit-wsinstance-http" = {
+      description = "Cockpit Web Service http instance";
+      bindsTo = [ "cockpit.service" ];
+      path = [ cfg.package ];
+      documentation = [ "man:cockpit-ws(8)" ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/libexec/cockpit-ws --no-tls --port=0";
+        User = "root";
+        Group = "";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit.socket
+    systemd.sockets."cockpit" = {
+      unitConfig = {
+        Description="Cockpit Web Service Socket";
+        Documentation="man:cockpit-ws(8)";
+        Wants="cockpit-motd.service";
+      };
+      socketConfig = {
+        ListenStream=cfg.port;
+        ExecStartPost= [
+          "-${cfg.package}/share/cockpit/motd/update-motd \"\" localhost"
+          "-${pkgs.coreutils}/bin/ln -snf active.motd /run/cockpit/motd"
```

Comment:
This should be done by tmpfiles

Reply:
This is the upstream approach.

/run/cockpit/motd starts pointing to the motd that says to enable the socket, when the socket is enabled  it changes /run/cockpit/motd to the motd that says that cockpit is accesible from $url, if that socket is disabled again then that link goes back to the first motd.

In my experience using it in both my machines, I think we could just use the enabled motd because the socket is always enabled and by just accessing port 9090 it wakes up the server.

Reply:
BTW the initial state is created by using tmpfiles